### PR TITLE
RedstoneWireBlock should be fully mapped now

### PR DESCRIPTION
**Pull request by lenrik1589**

### CODE REVIEW

**Comment by lenrik1589** [comment]
I hope these mappings make sense to you all, who know more than i do

File: mappings/net/minecraft/block/RedstoneWireBlock.mapping [hunk 1]
```diff
@@ -7,8 +7,15 @@ CLASS net/minecraft/class_2457 net/minecraft/block/RedstoneWireBlock
 	FIELD field_11439 WIRE_CONNECTION_WEST Lnet/minecraft/class_2754;
 	FIELD field_11440 WIRE_CONNECTION_NORTH Lnet/minecraft/class_2754;
 	FIELD field_24413 DOT_SHAPE Lnet/minecraft/class_265;
+	FIELD field_24414 LINE_SHAPE Ljava/util/Map;
+	FIELD field_24415 LINE_SIDE_SHAPE Ljava/util/Map;
+	FIELD field_24416 outlineShapes Ljava/util/Map;
+	FIELD field_24466 WIRE_COLOR [Lnet/minecraft/class_1160;
```
Comment (lenrik1589) [inline mappings/net/minecraft/block/RedstoneWireBlock.mapping:13]: I spent so much time searching for this field for so long ;-;

**Review by lenrik1589 — COMMENTED** [review]
I was searching for WIRE_COLOR so long :cry:

File: mappings/net/minecraft/block/RedstoneWireBlock.mapping [hunk 2]
```diff
@@ -7,8 +7,15 @@ CLASS net/minecraft/class_2457 net/minecraft/block/RedstoneWireBlock
 	FIELD field_11439 WIRE_CONNECTION_WEST Lnet/minecraft/class_2754;
 	FIELD field_11440 WIRE_CONNECTION_NORTH Lnet/minecraft/class_2754;
 	FIELD field_24413 DOT_SHAPE Lnet/minecraft/class_265;
+	FIELD field_24414 LINE_SHAPE Ljava/util/Map;
+	FIELD field_24415 LINE_SIDE_SHAPE Ljava/util/Map;
```
Comment (liach) [inline mappings/net/minecraft/block/RedstoneWireBlock.mapping]: hmm, why is a "shape" a map than a voxel shape? If the map key is like axis then name it like `SHAPES` (plural)

File: mappings/net/minecraft/block/RedstoneWireBlock.mapping [hunk 3]
```diff
@@ -7,8 +7,15 @@ CLASS net/minecraft/class_2457 net/minecraft/block/RedstoneWireBlock
...
+	FIELD field_24416 outlineShapes Ljava/util/Map;
+	FIELD field_24466 WIRE_COLOR [Lnet/minecraft/class_1160;
```
Comment (liach) [inline mappings/net/minecraft/block/RedstoneWireBlock.mapping]:
```suggestion
	FIELD field_24466 WIRE_COLORS [Lnet/minecraft/class_1160;
```

File: mappings/net/minecraft/block/RedstoneWireBlock.mapping [hunk 4]
```diff
@@ -7,8 +7,15 @@ CLASS net/minecraft/class_2457 net/minecraft/block/RedstoneWireBlock
...
 	FIELD field_24733 dotState Lnet/minecraft/class_2680;
-	METHOD method_10477 getRenderConnectionType (Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;Lnet/minecraft/class_2350;)Lnet/minecraft/class_2773;
+	METHOD method_10477 getRenderConnectionSideType (Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;Lnet/minecraft/class_2350;)Lnet/minecraft/class_2773;
```
Comment (liach) [inline mappings/net/minecraft/block/RedstoneWireBlock.mapping]: What does this method do? Inserting a "side" in the name doesn't make it more clear. Also afaik now mojang doesn't do client specific render states any more but compute it on server and send it to client, so I doubt the "render" part of the name as well.

File: mappings/net/minecraft/block/RedstoneWireBlock.mapping [hunk 5]
```diff
@@ -25,34 +32,46 @@ CLASS net/minecraft/class_2457 net/minecraft/block/RedstoneWireBlock
 		ARG 1 state
 	METHOD method_10487 getWireColor (I)I
 		ARG 0 powerLevel
-	METHOD method_27840 (Lnet/minecraft/class_1922;Lnet/minecraft/class_2680;Lnet/minecraft/class_2338;)Lnet/minecraft/class_2680;
+	METHOD method_27840 getWorldView (Lnet/minecraft/class_1922;Lnet/minecraft/class_2680;Lnet/minecraft/class_2338;)Lnet/minecraft/class_2680;
```
Comment (liach) [inline mappings/net/minecraft/block/RedstoneWireBlock.mapping]: How does it transform the input world view?

**Review by liach — APPROVED** [review]
Should be fine. Also update your pr to merge into 1.16.2 release.

File: mappings/net/minecraft/block/enums/WireConnection.mapping [hunk 6]
```diff
@@ -1,5 +1,8 @@
 CLASS net/minecraft/class_2773 net/minecraft/block/enums/WireConnection
 	FIELD field_12685 name Ljava/lang/String;
+	FIELD field_12686 UP Lnet/minecraft/class_2773;
+	FIELD field_12687 NONE Lnet/minecraft/class_2773;
+	FIELD field_12689 SIDE Lnet/minecraft/class_2773;
```
Comment (YanisBft) [inline mappings/net/minecraft/block/enums/WireConnection.mapping:5]: Why did you map these? They are deobfuscated

File: mappings/net/minecraft/block/RedstoneWireBlock.mapping [hunk 7]
```diff
@@ -7,8 +7,15 @@ CLASS net/minecraft/class_2457 net/minecraft/block/RedstoneWireBlock
...
 	FIELD field_24413 DOT_SHAPE Lnet/minecraft/class_265;
+	FIELD field_24414 LINE_SIDE_SHAPES Ljava/util/Map;
+	FIELD field_24415 LINE_UP_SHAPES Ljava/util/Map;
+	FIELD field_24416 outlineShapes Ljava/util/Map;
+	FIELD field_24466 WIRE_COLORS [Lnet/minecraft/class_1160;
 	FIELD field_24733 dotState Lnet/minecraft/class_2680;
-	METHOD method_10477 getRenderConnectionType (Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;Lnet/minecraft/class_2350;)Lnet/minecraft/class_2773;
+	METHOD method_10477 getSideConnectionState (Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;Lnet/minecraft/class_2350;)Lnet/minecraft/class_2773;
+		ARG 1 world
+		ARG 2 position
```
Comment (YanisBft) [inline mappings/net/minecraft/block/RedstoneWireBlock.mapping:17]:
```suggestion
		ARG 2 pos
```

File: mappings/net/minecraft/block/RedstoneWireBlock.mapping [hunk 8]
```diff
@@ -25,34 +32,46 @@ CLASS net/minecraft/class_2457 net/minecraft/block/RedstoneWireBlock
 		ARG 1 state
 	METHOD method_10487 getWireColor (I)I
 		ARG 0 powerLevel
-	METHOD method_27840 (Lnet/minecraft/class_1922;Lnet/minecraft/class_2680;Lnet/minecraft/class_2338;)Lnet/minecraft/class_2680;
+	METHOD method_27840 getBlockView (Lnet/minecraft/class_1922;Lnet/minecraft/class_2680;Lnet/minecraft/class_2338;)Lnet/minecraft/class_2680;
```
Comment (YanisBft) [inline mappings/net/minecraft/block/RedstoneWireBlock.mapping:35]: This method is wrongly named

File: mappings/net/minecraft/block/RedstoneWireBlock.mapping [hunk 9]
```diff
@@ -25,34 +32,46 @@ CLASS net/minecraft/class_2457 net/minecraft/block/RedstoneWireBlock
...
 		ARG 1 world
 		ARG 2 state
 		ARG 3 pos
+	METHOD method_27841 getSideConnectionState (Lnet/minecraft/class_1922;Lnet/minecraft/class_2338;Lnet/minecraft/class_2350;Z)Lnet/minecraft/class_2773;
+		ARG 1 world
+		ARG 2 position
```
Comment (YanisBft) [inline mappings/net/minecraft/block/RedstoneWireBlock.mapping:41]:
```suggestion
		ARG 2 pos
```

File: mappings/net/minecraft/block/RedstoneWireBlock.mapping [hunk 10]
```diff
@@ -25,34 +32,46 @@ CLASS net/minecraft/class_2457 net/minecraft/block/RedstoneWireBlock
...
+		ARG 3 direction
+		ARG 4 isTransparentAbove
```
Comment (YanisBft) [inline mappings/net/minecraft/block/RedstoneWireBlock.mapping:43]:
```suggestion
		ARG 4 transparentAbove
```

File: mappings/net/minecraft/block/RedstoneWireBlock.mapping [hunk 11]
```diff
@@ -25,34 +32,46 @@ CLASS net/minecraft/class_2457 net/minecraft/block/RedstoneWireBlock
...
 	METHOD method_27842 getReceivedRedstonePower (Lnet/minecraft/class_1937;Lnet/minecraft/class_2338;)I
 		ARG 1 world
 		ARG 2 pos
-	METHOD method_27843 (Lnet/minecraft/class_1922;Lnet/minecraft/class_2680;Lnet/minecraft/class_2338;)Lnet/minecraft/class_2680;
+	METHOD method_27843 getConnectionState (Lnet/minecraft/class_1922;Lnet/minecraft/class_2680;Lnet/minecraft/class_2338;)Lnet/minecraft/class_2680;
 		ARG 1 world
 		ARG 2 state
 		ARG 3 pos
-	METHOD method_27844 (Lnet/minecraft/class_1937;Lnet/minecraft/class_2338;)V
+	METHOD method_27844 provideNeighborUpdates (Lnet/minecraft/class_1937;Lnet/minecraft/class_2338;)V
```
Comment (YanisBft) [inline mappings/net/minecraft/block/RedstoneWireBlock.mapping:51]:
```suggestion
	METHOD method_27844 updateAllNeighbors (Lnet/minecraft/class_1937;Lnet/minecraft/class_2338;)V
```

File: mappings/net/minecraft/block/RedstoneWireBlock.mapping [hunk 12]
```diff
@@ -25,34 +32,46 @@ CLASS net/minecraft/class_2457 net/minecraft/block/RedstoneWireBlock
...
 		ARG 1 world
 		ARG 2 pos
-	METHOD method_27845 (Lnet/minecraft/class_2680;)Lnet/minecraft/class_265;
+	METHOD method_27845 makeOutlineShape (Lnet/minecraft/class_2680;)Lnet/minecraft/class_265;
```
Comment (YanisBft) [inline mappings/net/minecraft/block/RedstoneWireBlock.mapping:54]:
```suggestion
	METHOD method_27845 createOutlineShape (Lnet/minecraft/class_2680;)Lnet/minecraft/class_265;
```

File: mappings/net/minecraft/block/RedstoneWireBlock.mapping [hunk 13]
```diff
@@ -25,34 +32,46 @@ CLASS net/minecraft/class_2457 net/minecraft/block/RedstoneWireBlock
...
 		ARG 1 state
 	METHOD method_27846 isFullyConnected (Lnet/minecraft/class_2680;)Z
 		ARG 0 state
-	METHOD method_27936 (Lnet/minecraft/class_1937;Ljava/util/Random;Lnet/minecraft/class_2338;Lnet/minecraft/class_1160;Lnet/minecraft/class_2350;Lnet/minecraft/class_2350;FF)V
+	METHOD method_27936 emitDustParticle (Lnet/minecraft/class_1937;Ljava/util/Random;Lnet/minecraft/class_2338;Lnet/minecraft/class_1160;Lnet/minecraft/class_2350;Lnet/minecraft/class_2350;FF)V
 		ARG 1 world
 		ARG 2 random
 		ARG 3 pos
+		ARG 4 particleColor
```
Comment (YanisBft) [inline mappings/net/minecraft/block/RedstoneWireBlock.mapping:62]: A Vector3f can't be a particle color

Comment (liach) [reply mappings/net/minecraft/block/RedstoneWireBlock.mapping:62]: Hmm, I think mojang usese 3 floats for color in some places

**Comment by modmuss50** [comment]
Closing this due to no progress, if anyone wants to save this please open a new PR.

**Pull request closed without merging**